src/locked_mem.rs

```diff
@@ -182,4 +190,37 @@ mod tests {
 
         assert_eq!(8, call_me(&*locked.lock()));
     }
+
+    #[test]
+    fn clears_input_buffer_from_vec() {
+        let mut input = vec![1, 2, 3];
+        input.resize(3, 0);
+
+        let ptr = input.as_ptr();
+
+        let mut locked = LockedArray::from(input);
+
+        let vec = unsafe { Vec::from_raw_parts(ptr as *mut u8, 3, 3) };
```

Comment:
A little filthy in Rust :)

- Keep around a pointer to the input vector,
- Recreate a vector using that pointer after it should have been cleared
- Allocate nothing (not even the `vec![0, 0, 0]`) before reading that value back
- Forget the allocated vector to avoid double freeing the memory

It works, it means this is tested, possibly fragile.

Reply:
This is indeed fragile and UB according to rust's safety rules. I'm personally not too worried about testing zeroing the buffers... Consider dropping these tests and using Zeroize crate in those functions.